jpa/subsystem/src/main/java/org/jboss/as/jpa/config/PersistenceUnitMetadataImpl.java

```diff
@@ -201,16 +199,6 @@ public URL getPersistenceUnitRootUrl() {
         return persistenceUnitRootUrl;
     }
 
-    @Override
-    public void setAnnotationIndex(Map<URL, Index> indexes) {
-        annotationIndex = indexes;
-    }
-
-    @Override
-    public Map<URL, Index> getAnnotationIndex() {
```

Comment:
I guess we could also remove the org.jipijapa.plugin.spi.PersistenceUnitMetadata.setAnnotationIndex + getAnnotationIndex methods as that is an SPI (for WildFly components) that can change if needed.  

Or we could avod the change by adding empty PersistenceUnitMetadata.setAnnotationIndex + getAnnotationIndex methods to the PersistenceUnitMetadata interface.  

Adding methods means we keep our technical debt longer.  Removing the methods is better but that changes the spi.  

I'm now leaning towards removing the technical debt by removing the PersistenceUnitMetadata.setAnnotationIndex + getAnnotationIndex methods.

@bstansberry do you have a preference?